presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedSystemAccessControl.java

```diff
@@ -485,11 +485,9 @@ public void checkCanShowTables(SystemSecurityContext context, CatalogSchemaName
     @Override
     public Set<SchemaTableName> filterTables(SystemSecurityContext context, String catalogName, Set<SchemaTableName> tableNames)
     {
-        if (!canAccessCatalog(context.getIdentity(), catalogName, READ_ONLY)) {
-            return ImmutableSet.of();
-        }
-
-        return tableNames;
+        return tableNames.stream()
+                .filter(tableName -> checkAnyTablePermission(context, new CatalogSchemaTableName(catalogName, tableName)))
```

Comment:
you need to also check if user has any full access to the schema or catalog

Reply:
`checkAnyTablePermission` checks catalog access internally.  FWIU there is no schema `access` rule, and only a schema `owner` rule.

Reply:
I understand now.  I changed this to be `isSchemaOwner() || checkAnyTablePermission()`, so schema owners can see all the table names.